crates/rolldown/src/stages/generate_stage/code_splitting.rs

```diff
@@ -397,7 +397,16 @@ impl GenerateStage<'_> {
       for symbol_ref in info
         .namespace_refs
         .iter()
-        .filter(|item| self.link_output.used_symbol_refs.contains(item))
+        .filter_map(|ns| {
```

Comment:
We can’t rely on whether the owner of an imported rec namespace is included in the final bundle to determine if it can be a target for merging a CJS namespace. A CommonJS package can be marked as sideEffects: false, but its importer might still have its own side effects.
See https://github.com/rolldown/rolldown/pull/7602 for an example.
We also can’t use `used_symbol_refs` to filter out unused rec namespaces, because a rec namespace is declared by stmt_info, and when we include a stmt_info, we only include the referenced_symbols of that stmt_info.
See: https://github.com/rolldown/rolldown/blob/48b0d567d142afe6f00b0f11b03e164f4984a875/crates/rolldown/src/stages/link_stage/tree_shaking/include_statements.rs#L808